src/vision/shiny_future/grace_writes_a_recursive_fn_with_auto_box.md

```diff
@@ -0,0 +1,119 @@
+# ✨ Shiny future stories: Grace writes a recursive function with auto-box
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "shiny future" story submitted as part of the brainstorming period. It is derived from what actual Rust users wish async Rust should be, and is meant to deal with some of the challenges that Async Rust programmers face today.
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as peoples needs and desires for async Rust may differ greatly, shiny future stories [cannot be wrong]. At worst they are only useful for a small set of people or their problems might be better solved with alternative solutions). Alternatively, you may wish to [add your own shiny vision story][htvsq]!
+
+## The story
+
+### Early days
+
+Grace is using async Rust for the first time. She is experimenting with async functions and she begins with her favorite "Hello, World" sort of application, Fibonacci:
+
+```rust
+async fn fibonacci(n: u32) -> u32 {
+    match n {
+        0 => 1,
+        1 => 1,
+        _ => fibonacci(n - 1) + fibonacci(n - 2),
```

Comment:
```suggestion
        _ => fibonacci(n - 1).await + fibonacci(n - 2).await,
```

Reply:
Or, I guess, maybe I'm confused - these awaits seem necessary, but then this code won't compile as-is.

Reply:
I agree, I think I just overlooked the awaits?